src/object_store/src/object/s3.rs

```diff
@@ -114,6 +119,43 @@ impl ObjectStore for S3ObjectStore {
             .await?;
         Ok(())
     }
+
+    async fn list(&self, prefix: &str) -> ObjectResult<Vec<ObjectMetadata>> {
+        let mut ret: Vec<ObjectMetadata> = vec![];
+        let mut next_continuation_token = None;
+        loop {
+            let mut request = self
+                .client
+                .list_objects_v2()
+                .bucket(&self.bucket)
+                .prefix(prefix);
+            if let Some(continuation_token) = next_continuation_token.take() {
```

Comment:
What does this `continuation_token` mean? Could we have come comments explaining the general logic?

Reply:
Fixed.